low-level/framework/utils/utility.py

```diff
@@ -79,3 +81,42 @@ def get_os(self):
             else:
                 return None
 
+    def create_file(self, path):
+        """Create a file at specified path"""
+        dir = path[:path.rindex("/")]
+        if not os.path.exists(dir):
+            os.makedirs(dir)
+        if not os.path.exists(path):
+            file = open(path, "w+")
+            file.close()
+
+    def read_data(self, file):
+        """Reads the data from the file and returns it as a string"""
+        if not os.path.isfile(file):
+            return str(None)
+
+        with open (file, "r") as datafile:
+            return str(datafile.read().replace('\n', ''))
+
+    def get_hostname(self):
+        try:
+            return socket.getfqdn()
+        except Exception as ex:
+            print(f"Got exception: {ex} when trying to get hostname using getfqdn().")
+        try:
+            from subprocess import run, PIPE
+            from re import findall
+
+            IP_CMD = "ip -f inet addr show scope global up | grep inet"
+            IP_REGEX = b'\\b(\\d{1,3}(?:\\.\d{1,3}){3})/\d{1,2}\\b'
+
+            ip_out = run(IP_CMD, stdout=PIPE, shell=True, check=True)
+            ip_list = re.findall(IP_REGEX, ip_out.stdout)
+            if ip_list:
+                return ip_list[0]
```

Comment:
can we logger.debug() hostname before any 'return'

Reply:
Done.